app/web/src/features/communities/events/EditEventPage.tsx

```diff
@@ -56,30 +59,32 @@ export default function EditEventPage() {
         .add(endTime.get("minute"), "minute")
         .toDate();
 
+      updateEventInput = {
+        eventId,
+        isOnline: data.isOnline,
+        title: data.dirtyFields.title ? data.title : undefined,
+        content: data.dirtyFields.content ? data.content : undefined,
+        photoKey: data.dirtyFields.eventImage ? data.eventImage : undefined,
+        startTime:
+          data.dirtyFields.startTime || data.dirtyFields.startDate
+            ? finalStartDate
+            : undefined,
+        endTime:
+          data.dirtyFields.endTime || data.dirtyFields.endDate
+            ? finalEndDate
+            : undefined,
+      };
+
       if (data.isOnline) {
-        updateEventInput = {
-          eventId,
-          isOnline: data.isOnline,
-          title: data.title,
-          content: data.content,
-          photoKey: data.eventImage,
-          startTime: finalStartDate,
-          endTime: finalEndDate,
-          link: data.link,
-        };
-      } else {
-        updateEventInput = {
-          eventId,
-          isOnline: data.isOnline,
-          title: data.title,
-          content: data.content,
-          photoKey: data.eventImage,
-          startTime: finalStartDate,
-          endTime: finalEndDate,
+        updateEventInput = Object.assign(updateEventInput, {
+          link: data.dirtyFields.link ? data.link : undefined,
+        });
```

Comment:
Out of curiosity, is there a reason for using `Object.assign` insead of `updateEventInput.link = ...`?